api/graphql/types/application/permissions.py

```diff
@@ -23,3 +30,26 @@ def has_node_permission(cls, instance: Application, user: AnyUser, filters: dict
     @classmethod
     def has_update_permission(cls, instance: Application, user: AnyUser, input_data: dict[str, Any]) -> bool:
         return can_modify_application(user, instance)
+
+
+class UpdateAllApplicationOptionsPermission(BasePermission):
+    @classmethod
+    def has_update_permission(cls, instance: Application, user: AnyUser, input_data: dict[str, Any]) -> bool:
+        if user.is_anonymous:
+            return False
+        if user.is_superuser:
+            return True
+        if has_general_permission(user, required_permission="can_handle_applications"):
+            return True
+
+        units = [
+            unit_id
+            for section in instance.application_sections.all().prefetch_related(
+                Prefetch(
+                    "reservation_unit_options",
+                    ReservationUnitOption.objects.all().select_related("reservation_unit__unit"),
+                )
+            )
+            for unit_id in section.reservation_unit_options.all().values_list("reservation_unit__unit__id", flat=True)
+        ]
+        return has_unit_permission(user, "can_handle_applications", units)
```

Comment:
This would allow updating all options with permission only spanning one their units.

Reply:
Indeed, fixed this and added tests.